tools/IoTEdgeDevOps/DevOpsLib/IoTEdgeAgent.cs

```diff
@@ -108,34 +108,13 @@ public bool Equals(IoTEdgeVstsAgent other)
                 return true;
             }
 
-            if (this.id != other.id ||
-                !string.Equals(this.name, other.name, StringComparison.Ordinal) ||
-                !string.Equals(this.version, other.version, StringComparison.Ordinal) ||
-                this.status != other.status ||
-                this.enabled != other.enabled)
-            {
-                return false;
-            }
-
-            if (this.systemCapabilities.Count != other.systemCapabilities.Count ||
-                this.userCapabilities.Count != other.userCapabilities.Count)
-            {
-                return false;
-            }
-
-            var matchedSystemCapabilities = this.systemCapabilities.Intersect(other.systemCapabilities);
-            if (matchedSystemCapabilities.Count() != this.systemCapabilities.Count)
-            {
-                return false;
-            }
-
-            var matchedUserCapabilities = this.userCapabilities.Intersect(other.userCapabilities);
-            if (matchedUserCapabilities.Count() != this.userCapabilities.Count)
-            {
-                return false;
-            }
-
-            return true;
+            return this.id == other.id &&
+                string.Equals(this.name, other.name, StringComparison.Ordinal) &&
+                string.Equals(this.version, other.version, StringComparison.Ordinal) &&
+                this.status == other.status &&
+                this.enabled == other.enabled &&
+                this.systemCapabilities.SetEquals(other.systemCapabilities) &&
+                this.userCapabilities.SetEquals(other.userCapabilities);
```

Comment:
Can this be replaced by comparing the hash from `GetHash()`?

Reply:
No.  Hash code of 2 instances can be the same but they may not be equal.